draft-ietf-tls-esni.md

```diff
@@ -365,11 +365,11 @@ ClientHelloInner.
 The payload of the extension has the following structure:
 
 ~~~~
-    enum { outer(0), inner(1) } ClientECHType;
+    enum { outer(0), inner(1) } ECHClientHelloType;
 
     struct {
-       ClientECHType type;
-       select (ClientECH.type) {
+       ECHClientHelloType type;
```

Comment:
Does this field need to be explicit?  Empty/not is a pretty good discriminator.

Reply:
This was a change made in https://github.com/tlswg/draft-ietf-tls-esni/pull/422. The reasoning is that 8446 requires extensions to be solicited by the client, meaning in order for there to be an ECH extension in HRR, there must be an ECH extension in CHInner.

Reply:
Yes, but the extension can be empty and achieve the same end.

Reply:
Yeah, whether or not we need the octet indicator was raised in a separate PR. I suggest we move this off to a separate issue if we want to change it.

Reply:
Agreed that empty or not is a fine discriminator, though I'm not sure how to encode that cleanly in the TLS presentation language.

Reply:
We switch on external variables in other contexts.  Continuing to think of the presentation language as a formal grammar is probably not helpful..